CMakeLists.txt

```diff
@@ -20,6 +20,29 @@ endif ()
 project(remill)
 cmake_minimum_required(VERSION 3.2)
 
+if(EMSCRIPTEN)
```

Comment:
Can this be defined under cmake/settings.cmake as `option(REMILL_EMSCRIPTEN "description")`?
This makes sure the option will show up as a configuration setting in the project (i.e. make edit_cache)

Reply:
Just to be certain are you suggesting replacing all instances of `EMSCRIPTEN` with `REMILL_EMSCRIPTEN` and then defining that in the `generate.sh` script?

I can certainly make this change, but it seems prone to error since if you try to build with the EMSCRIPTEN.cmake toolchain it will certainly fail without defining `REMILL_EMSCRIPTEN`, and visa versa. I haven't actually used make edit_cache before, but if the purpose is to change the cached variables on the fly it seems like that would break in this case since we defined the toolchain when we called cmake. I'll take your advice on this one however.

Reply:
I initially thought it could be turned on and off with a setting, thanks for explaining how it works! It seems like your approach is the most correct one